src/coreclr/nativeaot/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -716,6 +716,30 @@ REDHAWK_PALEXPORT bool REDHAWK_PALAPI PalStartBackgroundWork(_In_ BackgroundCall
     return st == 0;
 }
 
+REDHAWK_PALIMPORT bool REDHAWK_PALAPI PalSetCurrentThreadName(const char* name)
+{
+    if (!name)
+    {
+        return true;
+    }
+    const char* name_copy = nullptr;
+    const int MAX_THREAD_NAME_SIZE = 15;
+    char* name_copy = new (nothrow) char[MAX_THREAD_NAME_SIZE + 1];
+    if (!name_copy)
+    {
+        return false;
+    }
```

Comment:
```suggestion
    char name_copy[MAX_THREAD_NAME_SIZE + 1];
```
`MAX_THREAD_NAME_SIZE ` is very small. It is fine to use stack buffer.